src/main/java/org/opentripplanner/graph_builder/module/NearbyStopFinder.java

```diff
@@ -165,8 +183,24 @@ public NearbyStopFinder(Graph graph, double radiusMeters, boolean useStreets) {
                 if (targetVertex instanceof TransitStopVertex && state.isFinal()) {
                     stopsFound.add(StopAtDistance.stopAtDistanceForState(state, ((TransitStopVertex) targetVertex).getStop()));
                 }
+                if (OTPFeature.FlexRouting.isOn()
+                    && targetVertex instanceof StreetVertex
+                    && ((StreetVertex) targetVertex).flexStopLocations != null) {
+                   for (FlexStopLocation flexStopLocation : ((StreetVertex) targetVertex).flexStopLocations) {
+                        locationsMap.put(flexStopLocation, state);
+                    }
+                }
             }
         }
+
+        for (var locationStates : locationsMap.asMap().entrySet()) {
+            FlexStopLocation flexStopLocation = locationStates.getKey();
+            Collection<State> states = locationStates.getValue();
+            State min = Collections.min(states, (s1, s2) -> (int) (s1.walkDistance - s2.walkDistance));
```

Comment:
Select the vertex from all vertices that are reachable by taking the minimum walking distance